### PR TITLE
Remove archives after an `HTTP` download

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by segiddins** [comment]
Reported the 2.0.0 failure to travis

File: .travis.yml [hunk 1]
```diff
@@ -3,13 +3,19 @@ os:
 
 language: ruby
 
+branches:
```
Comment (DanToml) [inline .travis.yml]: is this accidental?

Comment (segiddins) [reply .travis.yml]: nah, just copying from all the other CocoaPods CI configs

Comment (DanToml) [reply .travis.yml]: I would rather us not do that, because getting CI to run before opening a PR is nice, but :woman_shrugging:

Comment (segiddins) [reply .travis.yml]: But then you have CI running twice for each PR commit.... I've always just opened WIP PRs

Comment (DanToml) [reply .travis.yml]: Fair :woman_shrugging: - :shipit: when CI is :green_apple:

Comment (segiddins) [reply .travis.yml]: I'm waiting on travis for that :/

**Comment by dnkoutso** [comment]
@segiddins as discussed offline, I rebased the branch and applied the changes. This should be mergeable now

File: lib/cocoapods-downloader/remote_file.rb [hunk 2]
```diff
@@ -123,6 +123,8 @@ def extract_with_type(full_filename, type = :zip)
             FileUtils.move(entry.children, target_path)
           end
         end
+
+        FileUtils.rm(unpack_from) if File.exist?(unpack_from)
```
Comment (dnkoutso) [inline lib/cocoapods-downloader/remote_file.rb]: the `scp_spec` circumvents the downloading causing the `file.zip` not to exist and make the tests fail.

Added the `if` statement to fix it instead, it should be harmless.

**Comment by dnkoutso** [comment]
Green!

**Comment by segiddins** [comment]
Thanks so much for taking this over! :ship:

**Pull request merged**